File: runtime/ruby/units/wasi/fd_fdstat_set_flags.rb
def wasi_fd_fdstat_set_flags(fd, flags)
  meta = @fd_meta[fd]
  io = @fds[fd]
  return ERRNO_BADF if io.nil? || io.is_a?(WasiDir) || meta.nil?
  # Store the fdflags word; APPEND is honored by fd_write (which seeks to
  # end when the bit is set), so clearing it here disables append mode.
  meta[2] = flags & 0x1f
  ERRNO_SUCCESS
end

File: runtime/java/units/wasi/fd_fdstat_set_flags.java
// Update an fd's open fdflags. Only APPEND is behaviorally honored:
// fd_write seeks to end before each write when Handle.append is set, so
// toggling APPEND here flips that. The remaining flags (DSYNC/RSYNC/SYNC,
// NONBLOCK) are stored so fd_fdstat_get reflects them but carry no distinct
// behavior in this runtime.
int wasi_fd_fdstat_set_flags(int fd, int flags) {
    Object e = fds.get(fd);
    if (!fds.containsKey(fd)) {
        return WASI_BADF;
    }
    FdMeta m = meta.get(fd);
    if (m != null) {
        m.fdflags = flags;
    }
    if (e instanceof Handle) {
        ((Handle) e).append = (flags & 0x1) != 0; // fdflags::APPEND
    }
    return WASI_OK;
}
